pywr/core.py

```diff
@@ -118,9 +132,22 @@ def __init__(self, solver=None, parameters=None):
 
         self.node = {}
         self.group = {}
+        self.scenarios = ScenarioCollection()
 
         self.reset()
 
+    @property
+    def number_of_timesteps(self):
+        return len(self.timestepper)
+
+    @property
+    def number_of_scenarios(self):
```

Comment:
`len(model.scenarios.combinations)` has some effort involved in computing the combinations, whereas the length is a simple product. Therefore that would require `model.scenarios.combinations` being an object itself?